lib/Widget/ForecastIo.php

```diff
@@ -502,29 +577,44 @@ public function getResource($displayId = 0)
         }
 
         // Handle the daily template (if its here)
-        if (stripos($body, '[dailyForecast]')) {
+        $dailySubs = '';
+        $matches = '';
+        preg_match_all('/\[dailyForecast.*?\]/', $body, $matches);
+        // Substitute
+        foreach ($matches[0] as $sub) {
+            $replace = str_replace('[', '', str_replace(']', '', $sub));
+            // Handling for date/time
+            $itterations = 7;
+            $offset = 0;
+            if (stripos($replace, '|') > -1) {
+                $quantity = explode('|', $replace);
+                $itterations = $quantity[1];
+                
+                if (count($quantity) > 1)
+                  $offset = $quantity[2];
+            }
+
             // Pull it out, and run substitute over it for each day
-            $dailySubs = '';
             // Substitute for every day (i.e. 7 times).
-            for ($i = 0; $i < 7; $i++) {
+            for ($i = $offset; $i < $itterations + $offset; $i++) {
                 $dailySubs .= $this->makeSubstitutions($foreCast['daily']['data'][$i], $dailyTemplate);
             }
-
             // Substitute the completed template
-            $body = str_replace('[dailyForecast]', $dailySubs, $body);
+            $body = str_replace($sub, $dailySubs, $body);
         }
 
         // Run replace over the main template
         $data['body'] = $this->makeSubstitutions($foreCast['currently'], $body);
 
-
         // JavaScript to control the size (override the original width and height so that the widget gets blown up )
         $options = array(
             'previewWidth' => $this->getSanitizer()->getDouble('width', 0),
             'previewHeight' => $this->getSanitizer()->getDouble('height', 0),
             'originalWidth' => $this->region->width,
             'originalHeight' => $this->region->height,
-            'scaleOverride' => $this->getSanitizer()->getDouble('scale_override', 0)
+            'scaleOverride' => $this->getSanitizer()->getDouble('scale_override', 0),
+            'widgetDesignWidth' => $this->getOption('widgetOriginalWidth'),
```

Comment:
It would be great if you could sanitize these so they are integers

`'widgetDesignWidth' => $this->getSanitizer()->int($this->getOption('widgetOriginalWidth')),`

That means they will end up in JS as integers, which I think will work more consistently across browsers.